### PR TITLE
Update/application elfs

### PR DESCRIPTION
**Pull request by lpascal-ledger**
This PR updates the side-loaded applications used for testing. They have been renamed with the commit hashes they were generated from. They currently point on top of current `develop` branches, or `master` if no `develop` was present (case for `XRP`). This is temporary, last version of binaries should be automatically fetched rather than being versionned in this repository.

Also added a test highlighting a fixed bug in the Ethereum application.
Removed a GitHub workflow which was redundant with current workflow, only it was running on a versionned version of the exchange app, which was even more meaningless.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,8 +23,16 @@ make delete
...
+The `./test` directory contains files for testing the app and its interactions with other apps. Make sure to put up to date binaries in `test/elfs/` before launching the tests.
+In a future update, all binaries will be fetched automatically from Github.
+
+For now, only the exchange binary needs to be generated. The following flags are needed for the tests to pass:
+
+```
+make TESTING=1 TEST_PUBLIC_KEY=1
+```
+
+To run the tests:
```
Comment (fbeutin-ledger) [inline README.md:35]: + add the fact that the exchange app binaries must be copied as `test/elfs/exchange_nanos.elf` and `test/elfs/exchange_nanox.elf`

File: .gitignore [hunk 2]
```diff
@@ -29,6 +29,7 @@ _ReSharper*/
 [Tt]est[Rr]esult*
 .vs/
 #Nuget packages folder
+test/elfs/*
```
[the remark this answers is not in the document]
Comment (lpascal-ledger) [reply .gitignore:32]: All already versioned files won't be ignored, `git` will keep track of any change on them. And I'm not keen on adding git-related directory boilerplate in the test fixture.
@apaillier-ledger, could you provide a third opinion on this?

**Review by bigspider — COMMENTED** [review]
Looks like I failed to send the few comments I had on my review the other day.

File: .gitignore (hunk 2 above)
[the remark this answers is not in the document]
Comment (lpascal-ledger) [reply .gitignore:32]: Or @bigspider!

File: test/src/test.fixture.js [hunk 3]
```diff
@@ -1,56 +1,78 @@
-import Zemu from '@zondax/zemu';
+import Zemu, {DeviceModel} from '@zondax/zemu';
+
+var dir = require('node-dir')
```
[the remark this answers is not in the document]
Comment (lpascal-ledger) [reply test/src/test.fixture.js:3]: yes

File: test/src/test.fixture.js [hunk 4]
```diff
@@ -1,56 +1,78 @@
...
+    'ethereum': 'Ethereum',
+    'litecoin': 'Litecoin',
+    'stellar': 'Stellar',
+    'xrp': 'XRP',
+    'tezos': '"Tezos Wallet"'
 };
 
-const Resolve = require('path').resolve;
+function get_applications(nano_variant: string) {
+    var applications = {};
```
[the remark this answers is not in the document]
Comment (lpascal-ledger) [reply test/src/test.fixture.js:42]: indeed

File: test/src/test.fixture.js [hunk 5]
```diff
@@ -1,56 +1,78 @@
...
+    for (let path of dir.files('./elfs/', {sync: true})) {
+        Object.entries(SIDELOADED_APPLICATIONS).forEach(
+            ([file_prefix, name]) => {
+                if (path.split('/')[1].startsWith(file_prefix) && path.includes(nano_variant)) {
```
[the remark this answers is not in the document]
Comment (lpascal-ledger) [reply test/src/test.fixture.js:46]: Indeed, incoming fix

[the remark this answers is not in the document]
Comment (lpascal-ledger) [reply test/src/test.fixture.js:46]: Indeed, incoming fix

File: .gitignore (hunk 2 above)
[the remark this answers is not in the document]
Comment (bigspider) [reply .gitignore:32]: why ignoring, though? Not sure I see the advantage. If other files end up here that are used during testing, should only ignore those imho.

File: test/src/fund.test.js [hunk 6]
```diff
@@ -102,5 +103,48 @@ test('[Nano S] Overflow values should be trimmed when funding', zemu("nanos", as
...
+test('[Nano S] Valid Ethereum funding transaction should be accepted', zemu(nano_environments[0], async (sim) => {
+    const swap = new Exchange(sim.getTransport(), TRANSACTION_TYPES.FUND);
+    const transactionId_base64: string = await swap.startNewTransaction();
+    const transactionId: Buffer = base64url.toBuffer(transactionId_base64);
+    console.log("transactionID %s", transactionId.toString('hex'));
+
+    await swap.setPartnerKey(fundPartnerSerializedNameAndPubKey);
+    await swap.checkPartner(DERSignatureOfFundPartnerNameAndPublicKey);
+
+    var tr = new proto.ledger_swap.NewFundResponse();
```
[the remark this answers is not in the document]
Comment (lpascal-ledger) [reply test/src/fund.test.js]: yes

File: .gitignore (hunk 2 above)
[the remark this answers is not in the document]
Comment (lpascal-ledger) [reply .gitignore]: Alright, tie break

**Pull request merged**